CMPUT404W23T05/Social-Distribution-Front-End
# Authentication frontend

Linked to CMPUT404W23T05/CMPUT404-project-socialdistribution#12
### Tasks
- [x] very basic login and front end pages
  - [x] link in router
- [x] send form data to backend using an HTTP client (perhaps Axios?)
- [x] Get a token back
- [x] a store for tokens (using pinia)
- [x] use authentication state to restrict access to certain pages
- [x] Retrieve author info upon login
  - [x] display username and profile image on navbar
- [x] ability to logout
- [x] Display login and signup errors
- [x] Redirect from login to home if already signed in
- [x] make the pages look ~~good~~ okay 

### Screenshots
![Signup](https://user-images.githubusercontent.com/38046772/223001903-c74d4caf-30d6-4c24-9880-ead7b3558af0.png)
![Login](https://user-images.githubusercontent.com/38046772/223001952-dff687bc-d4dd-4510-a548-963de71b9e20.png)
![Error example](https://user-images.githubusercontent.com/38046772/223002212-5f8cb5f7-b3e5-42ce-a3ed-e01b55d11fe1.png)

### Test cases (for future unit testing)
- Log in, log out
- Log in, log out, log back in
- Log in, go to a different website, return (should still be logged in as local storage persists)
